src/Delay/Delay.tsx

```diff
@@ -0,0 +1,20 @@
+import React, { useState, useCallback, FC, ReactNode } from 'react'
+
+import { useTimeout } from '../useTimeout'
+
+interface DelayProps {
+  wait: number
+  children: ReactNode
+}
+
+export const Delay: FC<DelayProps> = ({ wait, children }) => {
+  const [show, setShow] = useState(false)
+
+  const showChildren = useCallback(() => {
```

Comment:
I'm not sure if this optimisation makes sense tbh. We're not using the returned `showChildren` to optimise an expensive operation, so maybe the useCallback hook itself adds more overhead than we save (re-creating the function on each render).

Definitely, the code becomes less complex if we simplify this to

```js
const [show, setShow] = useState(false)

useTimeout(() => setShow(true), wait)
```

Reply:
Good point